ibm/service/vpc/resource_ibm_is_vpc_default_network_acl.go

```diff
@@ -0,0 +1,872 @@
+					rule[isNetworkACLRuleName] = *rulex.Name
+					rule[isNetworkACLRuleAction] = *rulex.Action
+					rule[isNetworkACLRuleIPVersion] = *rulex.IPVersion
+					rule[isNetworkACLRuleSource] = *rulex.Source
+					rule[isNetworkACLRuleDestination] = *rulex.Destination
+					rule[isNetworkACLRuleDirection] = *rulex.Direction
+					rule[isNetworkACLRuleICMP] = make([]map[string]int, 0, 0)
+					rule[isNetworkACLRuleTCP] = make([]map[string]int, 0, 0)
+					rule[isNetworkACLRuleUDP] = make([]map[string]int, 0, 0)
+				}
+			}
+			rules = append(rules, rule)
+		}
+	}
+	d.Set(isNetworkACLRules, rules)
+	controller, err := flex.GetBaseController(meta)
+	if err != nil {
+		return err
+	}
+	d.Set(flex.ResourceControllerURL, controller+"/vpc-ext/network/acl")
+	d.Set(flex.ResourceName, *nwacl.Name)
+	// d.Set(flex.ResourceCRN, *nwacl.Crn)
+	return nil
+}
+
+func resourceIBMISVPCDefaultNetworkACLUpdate(d *schema.ResourceData, meta interface{}) error {
+	id := d.Id()
+
+	name := ""
+	hasChanged := false
+
+	if d.HasChange(isNetworkACLName) {
+		name = d.Get(isNetworkACLName).(string)
+		hasChanged = true
+	}
+
+	err := nwaclVPCDefaultUpdate(d, meta, id, name, hasChanged)
+	if err != nil {
+		return err
+	}
+	return resourceIBMISVPCDefaultNetworkACLRead(d, meta)
+}
+
+func nwaclVPCDefaultUpdate(d *schema.ResourceData, meta interface{}, id, name string, hasChanged bool) error {
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return err
+	}
+	rules := d.Get(isNetworkACLRules).([]interface{})
+	if hasChanged {
+		updateNetworkACLOptions := &vpcv1.UpdateNetworkACLOptions{
+			ID: &id,
+		}
+		networkACLPatchModel := &vpcv1.NetworkACLPatch{
+			Name: &name,
+		}
+		networkACLPatch, err := networkACLPatchModel.AsPatch()
+		if err != nil {
+			return fmt.Errorf("[ERROR] Error calling asPatch for NetworkACLPatch: %s", err)
+		}
+		updateNetworkACLOptions.NetworkACLPatch = networkACLPatch
+		_, response, err := sess.UpdateNetworkACL(updateNetworkACLOptions)
+		if err != nil {
+			return fmt.Errorf("[ERROR] Error Updating Network ACL(%s) : %s\n%s", id, err, response)
+		}
+	}
+	if d.HasChange(isNetworkACLTags) {
+		oldList, newList := d.GetChange(isNetworkACLTags)
+		err := flex.UpdateGlobalTagsUsingCRN(oldList, newList, meta, d.Get(isNetworkACLCRN).(string), "", isUserTagType)
+		if err != nil {
+			log.Printf(
+				"Error on update of resource network acl (%s) tags: %s", d.Id(), err)
+		}
+	}
+	if d.HasChange(isNetworkACLAccessTags) {
+		oldList, newList := d.GetChange(isNetworkACLAccessTags)
+		err := flex.UpdateGlobalTagsUsingCRN(oldList, newList, meta, d.Get(isNetworkACLCRN).(string), "", isAccessTagType)
+		if err != nil {
+			log.Printf(
+				"Error on update of resource network acl (%s) access tags: %s", d.Id(), err)
+		}
+	}
+	if d.HasChange(isNetworkACLRules) {
+		err := validateVPCDefaultInlineRules(rules)
+		if err != nil {
+			return err
+		}
+		//Delete all existing rules
+		err = clearVPCDefaultRules(sess, id)
```

Comment:
can we modify rules in-place and/or add/remove rules separately instead deleting existing rules ?